sktime/utils/results_writing.py

```diff
@@ -80,6 +109,144 @@ def write_results_to_uea_format(path, strategy_name, dataset_name, y_true,
     file.close()
 
 
+def write_dataframe_to_tsfile(
+    data,
+    path,
+    problem_name="sample_data",
+    timestamp=False,
+    univariate=True,
+    class_label=None,
+    class_value_list=None,
+    equal_length=False,
+    series_length=-1,
+    missing_values="NaN",
+    comment=None,
+):
+    """
+    Output a dataset in dataframe format to .ts file
+    Parameters
+    ----------
+    data: pandas dataframe
+        the dataset in a dataframe to be written as a ts file
+        which must be of the structure specified in the documentation
+        https://github.com/whackteachers/sktime/blob/master/examples/loading_data.ipynb
+        index |   dim_0   |   dim_1   |    ...    |  dim_c-1
+           0  | pd.Series | pd.Series | pd.Series | pd.Series
+           1  | pd.Series | pd.Series | pd.Series | pd.Series
+          ... |    ...    |    ...    |    ...    |    ...
+           n  | pd.Series | pd.Series | pd.Series | pd.Series
+    path: str
+        The full path to output the ts file
+    problem_name: str
+        The problemName to print in the header of the ts file
+        and also the name of the file.
+    timestamp: {False, bool}, optional
+        Indicate whether the data contains timestamps in the header.
+    univariate: {True, bool}, optional
+        Indicate whether the data is univariate or multivariate in the header.
+        If univariate, only the first dimension will be written to file
+    class_label: {list, None}, optional
+        Provide class label to show the possible class values
+        for classification problems in the header.
+    class_value_list: {list/ndarray, []}, optional
+        ndarray containing the class values for each case in classification problems
+    equal_length: {False, bool}, optional
+        Indicate whether each series has equal length. It only write to file if true.
+    series_length: {-1, int}, optional
+        Indicate each series length if they are of equal length.
+        It only write to file if true.
+    missing_values: {NaN, str}, optional
+        Representation for missing value, default is NaN.
+    comment: {None, str}, optional
+        Comment text to be inserted before the header in a block.
+
+    Returns
+    -------
+    None
+
+    Notes
+    -----
+    This version currently does not support writing timestamp data.
+
+    References
+    ----------
+    The code for writing series data into file is adopted from
+    https://stackoverflow.com/questions/37877708/
+    how-to-turn-a-pandas-dataframe-row-into-a-comma-separated-string
+    """
+    if class_value_list is None:
+        class_value_list = []
+    # ensure data provided is a dataframe
+    if not isinstance(data, pd.DataFrame):
```

Comment:
Could we use the input validation functions from `sktime.utils.validation.panel`, i.e. `check_X` with `coerce_to_pandas=True` so that this also works when passing a 3d numpy array?

Reply:
yeah sure, I wasn't aware there is already such module exist